src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -71,8 +72,8 @@ export default class DefaultDeviceController
   private readonly useWebAudio: boolean = false;
   private readonly useMediaConstraintsFallback: boolean = true;
 
-  private isStartAudioInputInProgress: boolean = false;
-  private isStartVideoInputInProgress: boolean = false;
+  private audioInputQueue: PromiseQueue = new PromiseQueue();
```

Comment:
nit: This naming might be confusing. How about something like below?:
`audioInputQueue` - >`audioInputTaskQueue`
`startAudioInputInner` -> `startAudioInputTask`
`stopAudioInputInner` -> `stopAudioInputTask`

Reply:
Nice suggestion. Updated.